crates/core/src/transport/connection_handler.rs

```diff
@@ -343,8 +345,35 @@ impl UdpPacketsListener {
                         continue;
                     }
                 }
+                ConnectionState::AckConnection => {
+                    let acknowledgment = SymmetricMessage::ack_ok(
+                        outbound_sym_key.as_mut().unwrap(),
+                        &self.this_peer_keypair.public,
+                    )?;
+                    let _ = self
+                        .socket
+                        .send_to(acknowledgment.data(), remote_addr)
+                        .await;
+                    // we are connected to the remote and we just send the pub key to them
+                    // if they fail to receive it, they will re-request the packet through
```

Comment:
Why are we sending the public key to them? Are you referring to the intro packet that contains the symmetric key encrypted with the other peer's public key? If so, that packet isn't part of the receipt mechanism implemented in the send/receive trackers.

Reply:
I will reverse the question: why shouldn't we send it? It IS already part of the protocol since is used for negotiating. 

Not sending it just makes it so we need to send it anyways with extra hops and complications. I would rather simplify the whole thing

Reply:
Plus, if you were to extract this code as a third party protocol (which maybe is what we should be doing in the future), how are you supposed to use it painlessly? I don't really see any benefits for removing this from here and have to request it implicitly anyways because is already part of the protocol.

Reply:
> I will reverse the question: why shouldn't we send it?

Because we don't need to send it for the transport module to establish a connection. Either it's two firewalled peers connecting, in which case both already have the other's public key, or it's a firewalled peer connecting to a gateway in which case the firewalled peer already has the gateway's public key, and the gateway doesn't need the firewalled peer's public key to talk to it - by design.

The public key is needed by a gateway as part of the `connect` protocol but that's a layer above the transport mechanism, you seem to be mixing these layers which I don't think will lead to greater simplicity.